ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/BlobSidecarValidator.java

```diff
@@ -0,0 +1,219 @@
+import tech.pegasys.teku.infrastructure.collections.LimitedSet;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.config.SpecConfigDeneb;
+import tech.pegasys.teku.spec.constants.Domain;
+import tech.pegasys.teku.spec.datastructures.blocks.SlotAndBlockRoot;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.deneb.SignedBlobSidecar;
+import tech.pegasys.teku.spec.datastructures.execution.versions.deneb.BlobSidecar;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult;
+
+/**
+ * This class supposed to implement gossip validation rules as per <a
+ * href="https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/p2p-interface.md#the-gossip-domain-gossipsub">spec</a>
+ */
+public class BlobSidecarValidator {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Spec spec;
+  private final Set<SlotAndBlockRoot> receivedValidBlobSidecarInfoSet;
+  private final GossipValidationHelper gossipValidationHelper;
+  final Map<Bytes32, BlockImportResult> invalidBlockRoots;
+
+  public static BlobSidecarValidator create(
+      final Spec spec,
+      final Map<Bytes32, BlockImportResult> invalidBlockRoots,
+      final GossipValidationHelper validationHelper) {
+
+    final Optional<Integer> maybeMaxBlobsPerBlock =
+        spec.forMilestone(spec.getForkSchedule().getHighestSupportedMilestone())
+            .getConfig()
+            .toVersionDeneb()
+            .map(SpecConfigDeneb::getMaxBlobsPerBlock);
+
+    final int validInfoSize = VALID_BLOCK_SET_SIZE * maybeMaxBlobsPerBlock.orElse(1);
+
+    return new BlobSidecarValidator(
+        spec, invalidBlockRoots, validationHelper, LimitedSet.createSynchronized(validInfoSize));
+  }
+
+  @VisibleForTesting
+  Set<SlotAndBlockRoot> getReceivedValidBlobSidecarInfoSet() {
+    return receivedValidBlobSidecarInfoSet;
+  }
+
+  private BlobSidecarValidator(
+      final Spec spec,
+      final Map<Bytes32, BlockImportResult> invalidBlockRoots,
+      final GossipValidationHelper gossipValidationHelper,
+      final Set<SlotAndBlockRoot> receivedValidBlobSidecarInfoSet) {
+    this.spec = spec;
+    this.invalidBlockRoots = invalidBlockRoots;
+    this.gossipValidationHelper = gossipValidationHelper;
+    this.receivedValidBlobSidecarInfoSet = receivedValidBlobSidecarInfoSet;
+  }
+
+  public SafeFuture<InternalValidationResult> validate(final SignedBlobSidecar signedBlobSidecar) {
+    final BlobSidecar blobSidecar = signedBlobSidecar.getBlobSidecar();
+
+    /*
+    [REJECT] The sidecar is for the correct topic -- i.e. sidecar.index matches the topic {index}.
+    */
+    /*
+    This rule is already implemented in
+    tech.pegasys.teku.networking.eth2.gossip.BlobSidecarGossipManager.TopicIndexAwareOperationProcessor
```

Comment:
Wondering if we should double check here...

Reply:
the problem is that apply the rule we need the topic name which is not exposed to the OperationProcessor. I think this is a very particular case and we can accept to just reference this as implemented internally by the gossip layer.
If there are others complaining about this let's further discuss :)